tests/suites/test_suite_ssl.function

```diff
@@ -2742,8 +2795,6 @@ void app_data_dtls(int mfl, int cli_msg_len, int srv_msg_len,
 
     test_app_data(mfl, cli_msg_len, srv_msg_len, expected_cli_fragments,
                   expected_srv_fragments, 1);
-    /* The goto below is used to avoid an "unused label" warning.*/
-    goto exit;
```

Comment:
Ah, it turns out I didn't consider one thing:
`exit` already existed here before, as it's added during test generation.
What decides on whether it's necessary or not is the usage in macros like `TEST_ASSERT`. 
The CI now complains about:
```
suites/test_suite_ssl.function:2708:1: warning: label ‘exit’ defined but not used [-Wunused-label]
 2708 | exit:
      | ^~~~
suites/test_suite_ssl.function: In function ‘test_app_data_tls’:
suites/test_suite_ssl.function:2784:1: warning: label ‘exit’ defined but not used [-Wunused-label]
 2784 | exit:
      | ^~~~
suites/test_suite_ssl.function: In function ‘test_app_data_dtls’:
suites/test_suite_ssl.function:2799:1: warning: label ‘exit’ defined but not used [-Wunused-label]
 2799 | exit:
 ```
 So please make sure that only the `goto exit;` lines for labels mentioned above stay.

Reply:
The test framework adds an `exit` label to all test function. Normally there's at least one call to `TEST_ASSERT` or a similar macro in the function, which contains `goto exit`. But occasionally all the ways a test can fail are via auxiliary functions, and so there's no failure assertion in the main test function, and so you need a `goto exit;` at the end of the function to keep compilers happy.

Reply:
My fault: I saw `MD_OR_USE_PSA_INIT` in those functions and then I assumed the `goto exit;` to be always there. However I didn't thought that this does not hold if the library is built without `MBEDTLS_MD_SOME_PSA` or `MBEDTLS_USE_PSA_CRYPTO`. I'll fix/revert that